### ISSUE TITLE
Snowflake: Multiple SQL statements in a single API call not supported

### ISSUE THREAD

**Issue by stephenahiggins** [TablePlus/TablePlus#3179]
### Please answer the questions below, it helps us to keep track of the issue. Please do not ignore it or your issue will be closed.

1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
Snowflake. AWS.
2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
524
3. The steps to reproduce this issue:
* Connect to a Snowflake instance 
* Select a database `USE DATABASE "northwind";`
* Import a CSV
* The schema is created successfully, but Snowflake rejects multiple INSERT INTO statements when inserting the content. 

<img width="649" alt="Screenshot 2023-12-17 at 21 12 47" src="https://github.com/TablePlus/TablePlus/assets/12612312/228defbe-722c-4dbe-8374-5982e4d38bf1">

Noted: If the bug is related to data, please attach an example SQL data.

**Comment by huyphams** [comment]
Hi @stephenahiggins I swap it into a transaction, if you have a chance, please try this patch:
https://www.dropbox.com/scl/fi/s0z4zqny5947eg2qhqw16/TablePlus.dmg?rlkey=76jzdex2woe2ib6epa3r6ibxs&dl=0
Hope it works now.

**Comment by stephenahiggins** [comment]
@huyphams it working, thanks! 

I've noticed a couple of other issues with the Snowflake connectivity. I've detailed them below. I can raise separate issues if that's better? 


1. I must enter `USE DATABASE` manually in the SQL before performing an action. When the database is selected in the UI, it isn't "selected" in the Snowflake session.
<img width="1252" alt="TablePlus1" src="https://github.com/TablePlus/TablePlus/assets/12612312/93a213d4-04c3-4af6-bfcf-c61f2998d546">

2. The tables are not shown when I select any schema. See screenshot, I have selected `PUBLIC`, but nothing is shown. I have tried refreshing. 
<img width="2141" alt="TablePlus2" src="https://github.com/TablePlus/TablePlus/assets/12612312/1b17883b-e68a-44cb-b151-2c5387a34977">
